algo/src/main/java/com/hartwig/serve/sources/ckb/region/CkbRegion.java

```diff
@@ -0,0 +1,25 @@
+package com.hartwig.serve.sources.ckb.region;
+
+import java.util.List;
+
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.Location;
+
+import org.immutables.value.Value;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Value.Immutable
+@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
+public abstract class CkbRegion {
+
+    @NotNull
+    public abstract String country();
+
+    @NotNull
+    public abstract List<String> states();
+
+    public boolean includes(@NotNull Location location) {
+        // TODO (PW): Shouldn't this be "states.isEmpty || states.contains()"? Or how would someone configure all states for a country?
+        return location.country().equals(country()) && (!states().isEmpty() && states().contains(location.state()));
```

Comment:
I was thinking that empty states = all states. It seems logical to me if you only specify a country, you get everything for that country. Otherwise you need to both add a special string and some error handling for empty states (which then will match nothing).

Reply:
That said, you're TODO was right that the logic was inverted :P

Reply:
I have no strong opinion, you could also make it null perhaps (null = all states) and rename the param to `specificStates`, though also fine with empty = all.

Reply:
I guess cleanest would be to have a predicate that is either alwaysTrue or filters on specific state :) but sounds like overkill